src/dcurl.c

```diff
@@ -93,14 +123,23 @@ void dcurl_destroy()
     struct list_head *p;
 
 #if defined(ENABLE_REMOTE)
-    destroyRemoteContext(&Remote_Context);
+    RemoteImplContext *remoteImpl = NULL;
+    struct list_head *pRemote;
+
+    list_for_each (pRemote, &REMOTE_IMPL_LIST) {
+        remoteImpl = list_entry(pRemote, RemoteImplContext, list);
+        destroyRemoteContext(remoteImpl);
+        list_del(pRemote);
+    }
 #endif
 
     list_for_each (p, &IMPL_LIST) {
         impl = list_entry(p, ImplContext, list);
         destroyImplContext(impl);
         list_del(p);
     }
+
+    initStatus = 0;
```

Comment:
Ditto. Avoid using raw `0` for reseting.

Reply:
Done.